### ISSUE TITLE
Koa support

### ISSUE THREAD

**Issue by niftylettuce** [alexindigo/agnostic#1]
It'd be great if we can get Koa support in here per https://github.com/agenda/agendash/issues/30 and <https://github.com/agenda/agendash/pull/45#issuecomment-271179975> cc @joeframbach

**Comment by alexindigo** [comment]
Agree, it would be awesome. Although I don't have much experience with Koa. Do you know what is lacking to make them work together? Thanks.

**Comment by joeframbach** [comment]
Hi Alex :) I am also not familiar with Koa. I'm just responding to Github issue requests and passing it forward.

**Comment by alexindigo** [comment]
Hi Joe. How are things? :)

I just noticed that issue is from 2016 :)
But in the same time, it's not exactly clear what is the original problem.

**Comment by OmgImAlexis** [comment]
I'd like to add koa support to this but it looks like the repo isn't really maintained anymore. If this is the case would you be okay with the agenda org taking this project over?

Ref: https://github.com/agenda

**Comment by alexindigo** [comment]
Hey, thanks for your interest. I'm still using the library in few of my projects.
How do you feel if we start with a PR and see how it goes from there.

**Comment by OmgImAlexis** [comment]
Is there any documentation on how to add a new adapter and/or a recommended way to do it?

**Comment by alexindigo** [comment]
This folder will expose you to the world of pain :) 
https://github.com/alexindigo/agnostic/tree/master/notes

Probably most straightforward way to approach it, would be to print out koa's Request / Response object (properties and methods) and see how it aligns with the existing API.

Most likely you'd need to change existing API and make it all async functions, so it would support both models. Then there is a question of that other servers – how they work with async functions/promises instead of callbacks. Seems like it'd be breaking change and most likely would require to drop support for old versions, in order to fully embrace the brave new world of async functions.

Good luck, thank you. And let me know if you hit a roadblock or there are questions.
